File: CleanAndroid/app/src/main/java/de/phillippbertram/cleanandroid/internal/di/scopes/PerApp.java
package de.phillippbertram.cleanandroid.internal.di.scopes;

import java.lang.annotation.Retention;
import javax.inject.Scope;

import static java.lang.annotation.RetentionPolicy.RUNTIME;

/**
 * A scoping annotation to permit objects whose lifetime should
 * conform to the life of application to be memorized in the
 * correct component.
 *
 * @author [PBE] Phillipp Bertram | phillipp.bertram@lmis.de
 */
@Scope
@Retention(RUNTIME)
public @interface PerApp {

}

File: CleanAndroid/app/src/main/java/de/phillippbertram/cleanandroid/internal/di/scopes/PerApp.kt
package de.phillippbertram.cleanandroid.internal.di.scopes


import javax.inject.Scope


/**
 * A scoping annotation to permit objects whose lifetime should
 * conform to the life of application to be memorized in the
 * correct component.

 * @author [PBE] Phillipp Bertram | phillipp.bertram@lmis.de
 */
@Scope
@Retention(AnnotationRetention.RUNTIME)
annotation class PerApp
